src/lib.rs

```diff
@@ -529,6 +619,34 @@ impl<'a, T: CBufItem, const SIZE: usize> CBufReader<'a, T, SIZE> {
         self.fetch_seq(idx, &())
     }
 
+    /// Tries to read the item at index `idx` in the buffer and return both a value and the index actually read.
+    ///
+    /// If successful, or if it had to skip forward to the earliest item, returns the item and its index.
+    /// Can fail due to index beyond end (`None`) or spinlock.
+    #[inline]
+    pub fn fetch_with_index(&self, idx: CBufIndex<SIZE>) -> FetchWithIndexResult<T, SIZE> {
+        let mut idx_try = idx;
+        for _ in 0..Self::NUM_READ_TRIES {
+            match self.fetch_seq(idx_try, &()) {
```

Comment:
Minor nitpick, but it would be better to use `CBufReader::fetch` rather than `CBufReader::fetch_seq` here.